lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -432,116 +355,70 @@ export class UserAgentApplication {
+    return new Promise<AuthResponse>((resolve, reject) => {
+      this.acquireTokenInteractive(Constants.interactionTypePopup, true, request, resolve, reject);
     });
   }
 
   /**
-   * @hidden
-   * @ignore
-   * Checks if the redirect response is received from the STS. In case of redirect, the url fragment has either id_token, access_token or error.
-   * @param {string} hash - Hash passed from redirect page.
-   * @returns {Boolean} - true if response contains id_token, access_token or error, false otherwise.
+   * Use when you want to obtain an access_token for your API via opening a popup window in the user's browser
+   * @param {@link AuthenticationParameters}
+   *
+   * To renew idToken, please pass clientId as the only scope in the Authentication Parameters
+   * @returns {Promise.<AuthResponse>} - a promise that is fulfilled when this function has completed, or rejected if an error was raised. Returns the {@link AuthResponse} object
    */
-  // TODO - rename this, the name is confusing
-  isCallback(hash: string): boolean {
-    hash = this.getHash(hash);
-    const parameters = Utils.deserialize(hash);
-    return (
-      parameters.hasOwnProperty(Constants.errorDescription) ||
-      parameters.hasOwnProperty(Constants.error) ||
-      parameters.hasOwnProperty(Constants.accessToken) ||
-      parameters.hasOwnProperty(Constants.idToken)
-    );
+  acquireTokenPopup(request: AuthenticationParameters): Promise<AuthResponse> {
+    return new Promise<AuthResponse>((resolve, reject) => {
+      this.acquireTokenInteractive(Constants.interactionTypePopup, false, request, resolve, reject);
+    });
   }
 
-  //#endregion
-
-  //#region Popup Flow
+  //#region Acquire Token
 
   /**
-   * Use when initiating the login process via opening a popup window in the user's browser
-   *
+   * Use when initiating the login process or when you want to obtain an access_token for your API,
+   * either by redirecting the user's browser window to the authorization endpoint or via opening a popup window in the user's browser.
    * @param {@link (AuthenticationParameters:type)}
    *
-   * @returns {Promise.<AuthResponse>} - a promise that is fulfilled when this function has completed, or rejected if an error was raised. Returns the {@link AuthResponse} object
+   * To renew idToken, please pass clientId as the only scope in the Authentication Parameters
    */
-  loginPopup(request?: AuthenticationParameters): Promise<AuthResponse> {
-    // Creates navigate url; saves value in cache; redirect user to AAD
-    return new Promise<AuthResponse>((resolve, reject) => {
-      // Fail if login is already in progress
-      if (this.loginInProgress) {
-        return reject(ClientAuthError.createLoginInProgressError());
-      }
+  private acquireTokenInteractive(interactionType: InteractionType, isLoginCall: boolean, request?: AuthenticationParameters, resolve?: any, reject?: any): void {
```

Comment:
Is there a way to do this without passing a boolean to clean up api a little bit?

Reply:
There isn't really a good way to do this since we need to identify which API was called internally. I understand that the internal function signature is a little messy, it will be cleaner when we refactor our internal promise implementation.

Reply:
sounds good